src/pad_tail.rs

```diff
@@ -48,7 +51,7 @@ where
     fn next(&mut self) -> Option<Self::Item> {
         match self.iter.next() {
             None => {
-                if self.pos < self.min {
+                if self.pos + self.back < self.total_len {
```

Comment:
I like this line, especially with my proposed refactor: `if self.elements_from_next + self.elements_from_next_back < self.elements_required`, then we generate another one.